api/v1alpha1/marin3r_types.go

```diff
@@ -0,0 +1,272 @@
+	if spec == nil {
+		new := &Marin3rSidecarSpec{}
+		new.Default(def)
+		return new
+	}
+	if !spec.IsDeactivated() {
+		copy := spec.DeepCopy()
+		copy.Default(def)
+		return copy
+	}
+	return spec
+}
+
+// +kubebuilder:validation:MinProperties:=1
+// +kubebuilder:validation:MaxProperties:=1
+type EnvoyDynamicConfig struct {
+	// ListenerHttp contains options for an HTTP/HTTPS listener
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	ListenerHttp *ListenerHttp `json:"listenerHttp,omitempty"`
+	// RouteConfiguration contains options for an Envoy route_configuration
+	// protobuffer message
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	RouteConfiguration *RouteConfiguration `json:"routeConfiguration,omitempty"`
+	// Cluster contains options for an Envoy cluster protobuffer message
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	Cluster *Cluster `json:"cluster,omitempty"`
+	// Runtime contains options for an Envoy runtime protobuffer message
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	Runtime *Runtime `json:"runtime,omitempty"`
+}
+
+type EnvoyDynamicConfigMeta struct {
+	// The name of the configuration/resource. The name is what
+	// allows a configuration to be used from wihin other configuration.
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	Name string `json:"name"`
+	// GeneratorVersion specifies the version of a given template.
+	// "v1" is the default.
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +kubebuilder:default:=v1
+	// +optional
+	GeneratorVersion *string `json:"generatorVersion,omitempty"`
+}
+
+// GetName returns the name
+func (meta *EnvoyDynamicConfigMeta) GetName() string {
+	return meta.Name
+}
+
+// GetGeneratorVersion returns the template's version
+func (meta *EnvoyDynamicConfigMeta) GetGeneratorVersion() string {
+	return *meta.GeneratorVersion
+}
+
+// EnvoyDynamicConfigRaw is a struct with methods to manage a
+// configuration defined using directly the Envoy config API
+type EnvoyDynamicConfigRaw struct {
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	// Allows defining configuration using directly envoy's config API.
+	// WARNING: no validation of this field's value is performed before
+	// writting the custom resource to etcd.
+	RawConfig *runtime.RawExtension `json:"rawConfig,omitempty"`
+}
+
+func (raw *EnvoyDynamicConfigRaw) GetRawConfig() []byte {
+	if raw != nil && raw.RawConfig != nil && raw.RawConfig.Raw != nil {
+		return raw.RawConfig.Raw
+	}
+	return nil
+}
+
+// ListenerHttp contains options for an HTTP/HTTPS listener
+type ListenerHttp struct {
+	EnvoyDynamicConfigMeta `json:",inline"`
+	EnvoyDynamicConfigRaw  `json:",inline"`
+	// The port where the listener listens for new connections
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// test comment
```

Comment:
comment to remove? Actually it is present on CRDs, docs...

Reply:
removed with 30ea36b

Reply:
Still 2 occurrences of `test comment`, **control-F** it  :stuck_out_tongue_winking_eye:

Reply:
nobody cares about the bundle files... :joy: 
fixed it in 7dd5293